xrpl/models/amounts/issued_currency_amount.py

```diff
@@ -25,3 +28,34 @@ class IssuedCurrencyAmount(IssuedCurrency):
 
     :meta hide-value:
     """
+
+    @classmethod
+    def from_issued_currency(
+        cls: Type[IssuedCurrencyAmount], issued_currency: Currency, value: str
+    ) -> Union[IssuedCurrencyAmount, str]:
```

Comment:
this doesn't belong in this class because it doesn't necessarily return an IssuedCurrencyAmount. Instead i'd rather see a method on `XRP` and `IssuedCurrency` to convert to an `Amount`.

Reply:
That doesn't work type-wise - there's a circular import dependency then